lib/sinon.js

```diff
@@ -7,7 +7,21 @@
  * Copyright (c) 2010-2013 Christian Johansen
  */
 "use strict";
-module.exports = require("./sinon/util/core");
+
+var match = require("./sinon/match");
+
+module.exports = exports = require("./sinon/util/core");
+
+exports.extend = require("./sinon/extend");
+exports.match = match;
+exports.typeOf = require("./sinon/typeOf");
+
+/*
+ * allow deepEqual to check equality of matchers through
+ * dependency injectection. Otherwise we get a circular
+ * dependency
+ */
+exports.deepEqual.match = match;
```

Comment:
Or I could require a function call to get deepEqual returned:

```js
module.exports = function(match) {
  return function deepEqual(a, b) {
    if (match && match.isMatcher(a, b)) {
  };
};
```

Imports would look like

```js
let deepEqualWithoutMatcher = require('.../core/deep-equal')();

let deepEqualWithMatcher = require('.../core/deep-equal')(require('sinon/match'));
```

This also breaks backcompat.